### PR TITLE
MCR-2444 Add more CLI options to manage access keys

### PR DESCRIPTION
**Pull request by golsch**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2444).

### CODE REVIEW

File: mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/MCRAccessKeyUtils.java [hunk 1]
```diff
@@ -109,6 +115,37 @@ public static synchronized void addAccessKeySecretToCurrentUser(final MCRObjectI
         addAccessKeySecret(MCRUserManager.getCurrentUser(), objectId, value);
     }
 
+    /**
+     * Cleans all access key secret attributes of users if the corresponding key does not exist.
+     */
+    public static void cleanUpUserAttributes() {
+        final List<MCRUser> users = MCRUserManager.listUsers(null, null, null, null);
```
Comment (rsteph-de) [inline mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/MCRAccessKeyUtils.java]: Does this scale for a large list of users (e.g. LDAP in Essen or Jena)

File: mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/cli/MCRAccessKeyCommands.java [hunk 2]
```diff
@@ -20,31 +20,135 @@
...
+        final MCRObjectID objectId = MCRObjectID.getInstance(objectIdString);
+        if (!MCRMetadataManager.exists(objectId)) {
+            throw new MCRException("MCRObject/Derivate doesn't exists.");
+        }
         MCRAccessKeyManager.clearAccessKeys(objectId);
-        LOGGER.info("cleared all access keys of {}.", objId);
+        LOGGER.info("Cleared all access keys of {}.", objectIdString);
+    }
+
+    private static MCRAccessKey readAccessKeyFromFile(final String pathString) throws IOException, MCRException {
```
Comment (rsteph-de) [inline mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/cli/MCRAccessKeyCommands.java]: static methods should go to top of class

File: mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/cli/MCRAccessKeyCommands.java [hunk 3]
```diff
@@ -20,31 +20,135 @@
...
+    }
+
+    @MCRCommand(syntax = "clean up access key user attributes",
+        help = "Cleans all access key secret attributes of users if the corresponding key does not exist.")
+    public static void cleanUp() {
+        MCRAccessKeyUtils.cleanUpUserAttributes();
+        LOGGER.info("Cleaned up access keys.");
+    }
+
+    @MCRCommand(syntax = "hash secret {0} for {1}",
```
Comment (rsteph-de) [inline mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/cli/MCRAccessKeyCommands.java]: maybe better: "hash access key secret ... for ..."

We might have other secrets, which I don't know yet.

**Comment by golsch** [comment]
Converted into draft because of #1349

**Review by rsteph-de — COMMENTED** [review]
some ideas

**Review by rsteph-de — APPROVED** [review]
looks good forme

**Pull request merged**